### PR TITLE
Clean up basic install script

### PR DESCRIPTION
**Pull request by haguenau**
**By submitting this pull request, I confirm the following:** 
*please fill any appropriate checkboxes, e.g: [X]*

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [x] I have made only one major change in my proposed changes.
- [x] I have commented my proposed changes within the code.
- [x] I have tested my proposed changes, and have included unit tests where possible.
- [x] I am willing to help maintain this change if there are issues with it later.
- [x] I give this submission freely and claim no ownership.
- [x] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [x] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

Please make sure you [Sign Off](https://github.com/pi-hole/pi-hole/wiki/How-to-signoff-your-commits.) all commits. Pi-hole enforces the [DCO](https://github.com/pi-hole/pi-hole/wiki/Contributing-to-the-project).

---
**What does this PR aim to accomplish?:**

This cleans up the Bash code of `basic-install.sh`, greatly reducing
the number of warnings emitted by ShellCheck, and in turn should make
it more likely that errors are caught in the future.

**How does this PR accomplish the above?:**

Mostly quoting variables. I did some very basic checking that the
script still runs. I don't know whether such cleanup changes are
welcome; feel free to ignore.

**What documentation changes (if any) are needed to support this PR?:**
*A detailed list of any necessary changes*

N/A.

### CODE REVIEW

**Comment by dschaper** [comment]
Thanks for the PR. We do welcome all submissions.

File: automated install/basic-install.sh [hunk 1]
```diff
@@ -233,19 +233,19 @@ if is_command apt-get ; then
...
     # Since our install script is so large, we need several other programs to successfully get a machine provisioned
     # These programs are stored in an array so they can be looped through later
-    INSTALLER_DEPS=(apt-utils dialog debconf dhcpcd5 git ${iproute_pkg} whiptail)
+    INSTALLER_DEPS=(apt-utils dialog debconf dhcpcd5 git "${iproute_pkg}" whiptail)
     # Pi-hole itself has several dependencies that also need to be installed
     PIHOLE_DEPS=(cron curl dnsutils iputils-ping lsof netcat psmisc sudo unzip wget idn2 sqlite3 libcap2-bin dns-root-data resolvconf libcap2)
     # The Web dashboard has some that also need to be installed
     # It's useful to separate the two since our repos are also setup as "Core" code and "Web" code
-    PIHOLE_WEB_DEPS=(lighttpd ${phpVer}-common ${phpVer}-cgi ${phpVer}-${phpSqlite})
+    PIHOLE_WEB_DEPS=(lighttpd "${phpVer}"-common "${phpVer}"-cgi "${phpVer}"-"${phpSqlite}")
```
Comment (Mcat12) [inline automated install/basic-install.sh]: The last two can be combined into one string:
```bash
"${phpVer}-${phpSqlite}"
```

File: automated install/basic-install.sh [hunk 2]
```diff
@@ -233,19 +233,19 @@ if is_command apt-get ; then
         phpVer="php$phpInsMajor.$phpInsMinor"
     fi
     # We also need the correct version for `php-sqlite` (which differs across distros)
-    if ${PKG_MANAGER} install --dry-run ${phpVer}-sqlite3 > /dev/null 2>&1; then
+    if "${PKG_MANAGER}" install --dry-run "${phpVer}"-sqlite3 > /dev/null 2>&1; then
```
Comment (Mcat12) [inline automated install/basic-install.sh]: The last argument can be combined into one string:
```
"${phpVer}-sqlite3"
```

File: automated install/basic-install.sh (hunk 1 above)
Comment (haguenau) [reply automated install/basic-install.sh]: Thanks. Addressed and squashed into the first commit.

File: automated install/basic-install.sh (hunk 2 above)
Comment (haguenau) [reply automated install/basic-install.sh]: If that's the preferred style, I will apply it to all instances I find.

Comment (haguenau) [reply automated install/basic-install.sh]: Addressed.

**Comment by PromoFaux** [comment]
There are a couple of merge conflicts to be resolved, but other than that it's looking good

**Comment by haguenau** [comment]
Alright, I just rebased on current development branch and eliminated conflicts.

**Comment by haguenau** [comment]
Thanks for merging!

**Comment by dschaper** [comment]
Thank you for submitting!

**Pull request merged**